src/Core/Services/GraphQLRequestValidator.cs

```diff
@@ -0,0 +1,538 @@
+                    runtimeConfig: runtimeConfig,
+                    columnsDerivedFromParentEntity: columnsDerivedFromParentEntity,
+                    columnsToBeDerivedFromEntity: columnsToBeDerivedFromEntity,
+                    nestingLevel: nestingLevel,
+                    parentEntityName: parentEntityName,
+                    sqlMetadataProviderFactory: sqlMetadataProviderFactory));
+            }
+        }
+
+        /// <summary>
+        /// Helper method to iterate over all the fields present in the input for the current field and
+        /// validate that the presence/absence of fields make logical sense for the multiple-create request.
+        /// </summary>
+        /// <param name="schemaObject">Input object type for the field.</param>
+        /// <param name="entityName">Current entity's name.</param>
+        /// <param name="context">Middleware Context.</param>
+        /// <param name="objectFieldNodes">List of ObjectFieldNodes for the the input field.</param>
+        /// <param name="runtimeConfig">Runtime config</param>
+        /// <param name="columnsDerivedFromSourceEntity">Set of columns in this entity whose values should be
+        /// derived from insertion in the source entity (i.e. source entity would have been the referenced entity).</param>
+        /// <param name="columnsToBeDerivedFromThisEntity">Set of columns in this entity whose values are to be
+        /// derived from insertion in the this entity and returned to the source entity so as to provide values for
+        /// the corresponding referencing fields (i.e. source entity would have been the referencing entity).</param>
+        /// <param name="nestingLevel">Current depth of nesting in the multiple-create request.</param>
+        /// <param name="parentEntityName">Parent entity's name.</param>
+        /// <param name="sqlMetadataProviderFactory">Metadata provider factory.</param>
+        private static void ValidateObjectFieldNodes(
+            InputObjectType schemaObject,
+            string entityName,
+            IMiddlewareContext context,
+            IReadOnlyList<ObjectFieldNode> objectFieldNodes,
+            RuntimeConfig runtimeConfig,
+            HashSet<string> columnsDerivedFromSourceEntity,
+            HashSet<string> columnsToBeDerivedFromThisEntity,
+            int nestingLevel,
+            string parentEntityName,
+            IMetadataProviderFactory sqlMetadataProviderFactory)
+        {
+            string dataSourceName = GraphQLUtils.GetDataSourceNameFromGraphQLContext(context, runtimeConfig);
+            ISqlMetadataProvider metadataProvider = sqlMetadataProviderFactory.GetMetadataProvider(dataSourceName);
+            SourceDefinition sourceDefinition = metadataProvider.GetSourceDefinition(entityName);
+            Dictionary<string, IValueNode?> backingColumnData = MultipleCreateOrderHelper.GetBackingColumnDataFromFields(context, entityName, objectFieldNodes, metadataProvider);
```

Comment:
`GetBackingColumnDataFromFields` -> Gets the backing column data from fields. The column data can only be obtained for fields which represent a column. Hence, in your example, when this method is called for Publisher, it will return `name`, for Book, it will return `title`.